Add initial Vue project setup #8

Komediruzecki opened the pull request:
Add all initial app files (Vue 3, Node LTS (20.x.x), Typescript) 
Add vite test suite for easier testing and lint/prettier

Test: Run development build

Komediruzecki committed:
Add initial Vue project setup #8

Add all initial app files (Vue 3, Node LTS (20.x.x), Typescript)
Add vite test suite for easier testing and lint/prettier

Test: Run development build

suncica-negra commented on tabu-app/index.html:
```diff
@@ -0,0 +1,13 @@
+    <meta charset="UTF-8">
+    <link rel="icon" href="/favicon.ico">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Vite App</title>
```
Možda bi ovdje mogli odmah saviti pravi naziv - Tabu 🙂

Komediruzecki replied:
Fixed

suncica-negra commented on tabu-app/public/favicon.ico:
Ako u figmi postoji odgovarajući favicon/logo od Tabu-a možda isto ne bi bilo loše odmah ga staviti umjesto ovog.

suncica-negra replied:
I can. Or is it a question for @Komediruzecki ?

Komediruzecki replied:
I'll update this before we merge

Komediruzecki replied:
Fixed

suncica-negra commented on tabu-app/src/App.vue:
```diff
@@ -0,0 +1,91 @@
+  <RouterView />
+</template>
+
+<style scoped>
```
Umjesto CSSa bilo bi dobro dodati SCSS. Ali to možda bolje prilikom prvog dodavanja 'stvarnih' stilova.

Komediruzecki replied:
I was thinking about something like this: https://tailwindcss.com/docs/guides/vite, have you tried this approach?

On previous projects I used SCSS, but most recently styled components in React, it seemed quite capable and easy to scale, however it might be a bit more complex than needed  https://styled-components.com/

suncica-negra replied:
I wouldn't say I like remembering many classes to add some style, so I don't like Tailwind and similar frameworks. 

Also, why introduce styles via 'harder' JS? I believe that the application will be 'lighter' with as little JS as possible, and that includes more (S)CSS and less JS. 

But if Tailwind and styled-components will be accepted as a basis for the project, I will adapt 🙂

Komediruzecki replied:
For the start of project it would be best to start small and see what works, so yeah, lets use SCSS, possibly there is also some neat way to handle vendor prefixes? If that does not work, we can consider other approaches

suncica-negra replied:
Ok, I agree 🙂 

Yes, there is https://github.com/postcss/autoprefixer that we can add to the project to take care of prefixes.

suncica-negra commented on tabu-app/src/assets/logo.svg:
Ako u figmi postoji odgovarajući logo od Tabu-a možda isto ne bi bilo loše odmah ga staviti umjesto ovog.

vatroslav-tabu replied:
It was created. Can you update this, then?

suncica-negra replied:
I can. Or is it a question for @Komediruzecki ?

Komediruzecki replied:
Updated logo

suncica-negra reviewed (commented):
Generalno bih rekla da je to za početak ok osim prijašnjih komentara. Ima dosta fajlova koje bude trebalo obrisati, ali to možemo prvom prilikom kad se krene raditi nešto konkretnoga 😃

Komediruzecki committed:
Update initial project setup in Vue

Replace logo and favicon with tabu images
Remove generated content and icons
Setup login and about page as a initial base
Update component test to check login page for string 'tabu'
Update index.html with Tabu name

ghost commented:
Since I don't understand this code, I'll merge the pull request when you two both agree on it. :)

suncica-negra reviewed (approved):
I would say that it is ok to begin with and that it can merge into master/main 🥳

Komediruzecki commented:
> I would say that it is ok to begin with and that it can merge into master/main 🥳

Great, @Vatroslav you can merge it.

Komediruzecki commented:
> Generalno bih rekla da je to za početak ok osim prijašnjih komentara. Ima dosta fajlova koje bude trebalo obrisati, ali to možemo prvom prilikom kad se krene raditi nešto konkretnoga 😃

Great, I've removed most of template files, hopefully this is good for first merge.
